### PR TITLE
add method to get metadata/list_groups

### PR DESCRIPTION
**Pull request by olegrok**
See https://github.com/edenhill/librdkafka/blob/cfc0731617ddd6858058f31f564772202f209e72/src/rdkafka.h#L4725

Syntax:
  consumer:metadata({timeout_ms = <timeout>})
  producer:metadata({timeout_ms = <timeout>}, topic = <topic(string)>)

Part of #59

### CODE REVIEW

**Review by amitichev — APPROVED** [review]

File: kafka/consumer.c [hunk 1]
```diff
@@ -788,3 +788,30 @@ lua_consumer_dump_conf(struct lua_State *L) {
...
 }
+
+int
+lua_consumer_metadata(struct lua_State *L) {
+    consumer_t **consumer_p = (consumer_t **)luaL_checkudata(L, 1, consumer_label);
+    if (consumer_p == NULL || *consumer_p == NULL)
+        return 0;
+
+    if ((*consumer_p)->rd_consumer != NULL) {
+        int timeout_ms = lua_tonumber(L, 2);
```
Comment (hackallcode) [inline kafka/consumer.c:799]: In `lua_consumer_list_groups` you are using an integer for the same timeout. Maybe it's better to use an integer here too.

File: kafka/init.lua [hunk 2]
```diff
@@ -203,6 +205,31 @@ function Consumer:dump_conf()
...
+
+    local timeout_ms = DEFAULT_TIMEOUT_MS
+    if options ~= nil and options.timeout_ms ~= nil then
+        timeout_ms = options.timeout_ms
+    end
+
+    return self._consumer:metadata(timeout_ms)
+end
+
+function Consumer:list_groups(group, timeout_ms)
```
Comment (hackallcode) [inline kafka/init.lua:221]: Maybe it's better to use options to configure timeout as in `Consumer:metadata` function in order to support one style

File: kafka/init.lua [hunk 3]
```diff
@@ -359,6 +386,36 @@ function Producer:dump_conf()
...
+        return
+    end
+
+    local timeout_ms = DEFAULT_TIMEOUT_MS
+    if options ~= nil and options.timeout_ms ~= nil then
+        timeout_ms = options.timeout_ms
+    end
+
+    local topic
+    if options ~= nil and options.topic ~= nil then
```
Comment (hackallcode) [inline kafka/init.lua:400]: Is it `box.NULL` protection? Why it's impossible to use options.topic as is?

File: kafka/init.lua [hunk 4]
```diff
@@ -359,6 +386,36 @@ function Producer:dump_conf()
...
+        topic = options.topic
+    end
+
+    return self._producer:metadata(topic, timeout_ms)
+end
+
+function Producer:list_groups(group, timeout_ms)
```
Comment (hackallcode) [inline kafka/init.lua:407]: The same comment: maybe it's better to use `options` style for `timeout` and probably `group` options

File: kafka/producer.c [hunk 5]
```diff
@@ -124,7 +124,7 @@ add_producer_topics(producer_topics_t *topics, rd_kafka_topic_t *element) {
     return 0;
 }
 
-rd_kafka_topic_t *
+static rd_kafka_topic_t *
```
Comment (hackallcode) [inline kafka/producer.c:127]: Why are you using the `static` word here?

File: kafka/producer.c [hunk 6]
```diff
@@ -656,3 +656,42 @@ lua_create_producer(struct lua_State *L) {
...
+    if (producer_p == NULL || *producer_p == NULL)
+        return 0;
+
+    if ((*producer_p)->rd_producer != NULL) {
+        rd_kafka_topic_t *topic = NULL;
+        const char *topic_name = lua_tostring(L, 2);
+        if (topic_name != NULL) {
+            topic = find_producer_topic_by_name((*producer_p)->topics, topic_name);
+            if (topic == NULL)
+            {
```
Comment (hackallcode) [inline kafka/producer.c:672]:
```suggestion
            if (topic == NULL) {
```

File: kafka/producer.c (hunk 5 above)
Comment (olegrok) [reply kafka/producer.c:127]: This function isn't used outside current file.

File: kafka/init.lua (hunk 3 above)
Comment (olegrok) [reply kafka/init.lua:406]: Consider it as box.NULL protection. Just to be sure.

File: kafka/producer.c (hunk 6 above)
Comment (olegrok) [reply kafka/producer.c]: Fixed. Thanks

File: kafka/init.lua (hunk 4 above)
Comment (olegrok) [reply kafka/init.lua]: Sounds reasonable. Fixed.

File: kafka/init.lua (hunk 2 above)
Comment (olegrok) [reply kafka/init.lua]: Fixed as well

**Review by hackallcode — APPROVED** [review]

**Pull request merged**